src/Data/Enum.purs

```diff
@@ -37,10 +37,12 @@ derive newtype instance ordCardinality :: Ord (Cardinality a)
 -- | Type class for enumerations.
 -- |
 -- | Laws:
--- | - `succ a > pred a`
--- | - `pred a < succ a`
--- | - `pred >=> succ >=> pred = pred`
--- | - `succ >=> pred >=> succ = succ`
+-- | - Successor: `maybe true (a < _) (succ a)`
+-- | - Predecessor: `maybe true (_ < a) (pred a)`
+-- | - Succ retracts pred: `pred a >>= succ >>= pred = pred a`
+-- | - Pred retracts succ: `succ a >>= pred >>= succ = succ a`
+-- | - Non-skipping succ: `b <= a || maybe false (_ <= b) (succ a)`
```

Comment:
yes, I'm not sure either. On the one hand, I have to unpack the `maybe` definitions when I read them, where I find the `any`/`all` definitions clearer (if I read `all` as `∀`, etc.), but `Foldable` does add some mental overhead.